bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -1286,6 +1294,9 @@ public void setData(String key, Object value) {
 			this.nativeZoom = getShellZoom();
 		}
 	}
+	if (PROPOGATE_AUTOSCALE_DISABLED.equals(key)) {
+		this.propagateAutoScaleDisabled = Boolean.parseBoolean(value.toString());
```

Comment:
Do we need to propagate the news setting to childs that may already exist? With the current implementation, childs will only be properly affected if `setData("PROPAGATE_AUTOSCALE_DISABLED")` is called before those children are created, isn't it? Or is that even intended, as when the first childs were created that flag was not set yet?

Reply:
The only currently planned usage is in GEF, where the data would be set directly in the constructor. This ensures that all children will see the flag set. Given our current use cases, the state appears fine. In my opinion, it is acceptable not to modify existing children created before the flag was introduced, in case the flag is used elsewhere.